examples/medplum-task-example-app/bots/src/create-respond-to-message-task.ts

```diff
@@ -0,0 +1,210 @@
+import { getReferenceString, MedplumClient, parseReference } from '@medplum/core';
+import { Bundle, BundleEntry, Communication, Practitioner, Reference, ResourceType, Task } from '@medplum/fhirtypes';
+
+/**
+ * This bot creates a task for any messages that have not been responded to in 30 minutes. It is set to run every 15 minutes, and
+ * only create one task per thread. If an employee has already responded to the thread, it will automatically be assigned to that
+ * employee, otherwise it will be assigned to the Care Coordinator queue.
+ *
+ * @param medplum - The medplum client
+ * @returns Promise<any>
+ */
+export async function handler(medplum: MedplumClient): Promise<any> {
+  const currentDate = new Date();
+  const thirtyMinutesAgo = new Date(currentDate.getTime() - 30 * 60 * 1000);
+  const timeStamp = thirtyMinutesAgo.toISOString();
+
+  // Get all messages that are part of an active thread and older than 30 minutes
+  const messages: Communication[] = await medplum.searchResources('Communication', {
+    sent: `lt${timeStamp}`,
+    'part-of:missing': false,
+    // 'part-of:Communication.status': 'in-progress',
+  });
+
+  // Filter for messages sent by patients
+  const patientMessages = messages.filter((message) => getMessageSenderType(message) === 'Patient');
+
+  // If no messages from patients, return
+  if (patientMessages.length === 0) {
+    console.log('No messages in the last 30 minutes that require a response.');
+    return false;
+  }
+
+  // Create and execute a batch search for all thread messages
+  const searchBundle = buildSearchBundle(patientMessages);
```

Comment:
rename `buildSearchBundle`. Maybe makes sense to change the helper function to encapsualte both the bundle createion + the `executeBatch`. 

Then you can call it something like `fetchMessageThreads`